test/rules/object-comparison/allow-object-equal-comparison/test.ts.lint

```diff
@@ -0,0 +1,147 @@
+if (2) {}
+if (2 > 1) {}
+if (2 < 1) {}
+if (2 >= 1) {}
+if (2 <= 1) {}
+if (2 == 1) {}
+if (2 === 1) {}
+if (2 != 1) {}
+if (2 !== 1) {}
+if (2 > undefined) {}
+    ~~~~~~~~~~~~~     [cannot compare type 'number' to type 'null | undefined']
```

Comment:
Can we use sentence case for these failure strings? let's capitalize the first word